spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/batch/TraceJobExecutionListener.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.batch;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionListener;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanAndScope;
+import org.springframework.cloud.sleuth.Tracer;
+
+class TraceJobExecutionListener implements JobExecutionListener {
+
+	private final Tracer tracer;
+
+	private static Map<JobExecution, SpanAndScope> SPANS = new ConcurrentHashMap<>();
+
+	TraceJobExecutionListener(Tracer tracer) {
+		this.tracer = tracer;
+	}
+
+	@Override
+	public void beforeJob(JobExecution jobExecution) {
+		Span span = this.tracer.nextSpan().name(jobExecution.getJobInstance().getJobName());
+		span.tag("job.name", jobExecution.getJobInstance().getJobName());
```

Comment:
@cppwfs @mminella In order to make Spring Batch more observable™, what do you think about having an actuator endpoint where you can read and search the contents of the JobRepository? :)

Seeing the available job names, their instances, their executions and their stepexecutions too through a nice REST API (batteries and HATEOAS™ are included). Basically creating view objects for the `Job`, `JobInstance`, `JobExecution`, and `StepExecution` abstractions queried from the `JobExplorer`.

Reply:
I will yield to @benas and @mminella experiece on this one.   I've not pushed for this feature because of the ephemeral nature of batch and Spring Cloud Task apps.   This information was obtainable from batch's database tables via SCDF, integration with Micrometer and, now Sleuth.

Reply:
@jonatan-ivanov +1 to @cppwfs 's comment. Given that most batch jobs run via Boot are ephemeral, having that app expose an API doesn't make much sense (would it be up when you wanted to query it?). That's what Spring Cloud Data Flow is for. It provides exactly what you are describing...a REST API that exposes the data in the `JobRepository`.

Reply:
I second what Glenn and Michael said. For the record, this has been already discussed in boot here: https://github.com/spring-projects/spring-boot/issues/21024.